MAVProxy/modules/mavproxy_steamvr.py

```diff
@@ -0,0 +1,303 @@
+            [pitch,roll,yaw] = self.tracker_q_to_euler(q[0],q[1],q[2],q[3])
+            pitch *= -1
+            yaw *= -1
+            yaw += math.radians(self.vr_settings.yaw_offset + 180)
+            yaw = math.radians(mavextra.wrap_360(math.degrees(yaw)))
+
+            # we also have acesees to the rotation rates in rads/s
+            # I have not worked out the conversion as it is not currently used
+            #rot_rate = pose.vAngularVelocity
+
+            self.pos = pos_ned
+            self.att = [math.degrees(roll), math.degrees(pitch), math.degrees(yaw)]
+
+            # read in velocity's
+            vel_in = pose.vVelocity
+            vel_ned = Vector3(-vel_in[2], vel_in[0], -vel_in[1])
+            self.vel = vel_ned
+
+            # match requested send rate
+            if self.vr_settings.vision_rate > 0:
+                dt = now - last_msg_time
+                if dt < 1.0 / self.vr_settings.vision_rate:
+                    continue
+            last_msg_time = now
+
+            mav = self.master
+            time_us = int(now * 1.0e6)
+            if now - last_origin_send > 1 and self.vr_settings.vision_rate > 0:
+                # send a heartbeat msg
+                mav.mav.heartbeat_send(mavutil.mavlink.MAV_TYPE_GCS, mavutil.mavlink.MAV_AUTOPILOT_GENERIC, 0, 0, 0)
+
+                # send origin at 1Hz
+                mav.mav.set_gps_global_origin_send(self.target_system,
+                                                   int(self.vr_settings.origin_lat*1.0e7),
+                                                   int(self.vr_settings.origin_lon*1.0e7),
+                                                   int(self.vr_settings.origin_alt*1.0e3),
+                                                   time_us)
+                last_origin_send = now
+
+            if self.vr_settings.gps_rate > 0:
+                gps_period = 1 / self.vr_settings.gps_rate
+                if now - last_gps_send > gps_period:
+                    '''send GPS data at the specified rate, trying to align on the given period'''
+                    yaw_cd = int(math.degrees(yaw) * 100)
+                    if yaw_cd == 0:
+                        # the yaw extension to GPS_INPUT uses 0 as no yaw support
+                        yaw_cd = 36000
+
+                    gps_lat, gps_lon = mavextra.gps_offset(self.vr_settings.origin_lat,
+                                                        self.vr_settings.origin_lon,
+                                                        pos_ned.y, pos_ned.x)
+                    gps_alt = self.vr_settings.origin_alt - pos_ned.z
+
+                    gps_week, gps_week_ms = get_gps_time(now)
+
+                    if self.vr_settings.gps_nsats >= 6:
+                        fix_type = 3
+                    else:
+                        fix_type = 1
+                    mav.mav.gps_input_send(time_us, 0, 0, gps_week_ms, gps_week, fix_type,
```

Comment:
Why does gps_input need to be sent if you're also sending global_vision_position_estimate?

Reply:
It just gives the option, by default the gps stream rate is 0